### PR TITLE
SO-2918: Support MRCM relationship concept set definitions

### PR DESCRIPTION
**Pull request by AAAlinaaa**

### CODE REVIEW

File: snomed/com.b2international.snowowl.snomed.datastore/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -27,7 +27,8 @@ Require-Bundle: org.eclipse.core.runtime;bundle-version="3.9.0",
  com.b2international.mapdb;bundle-version="[3.0.5,3.0.6)",
  com.fasterxml.jackson.dataformat.jackson-dataformat-csv;bundle-version="2.8.6",
  com.b2international.snowowl.core,
- org.eclipse.core.resources
+ org.eclipse.core.resources,
+ com.b2international.commons
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.datastore/META-INF/MANIFEST.MF]: This is already a dependency via `snowowl.core`, please remove it.

File: snomed/com.b2international.snowowl.snomed.api.rest.tests/src/com/b2international/snowowl/snomed/api/japi/io/SnomedRefSetDSVExportTest.java [hunk 2]
```diff
@@ -180,7 +181,14 @@ public void mapTypeDSVExport() throws Exception {
...
 	private Iterable<SnomedConstraint> getConstraints(SnomedConcepts concepts) {
 		return SnomedRequests
-				.prepareGetApplicablePredicates(MAIN_BRANCH, idsOf(concepts), ancestorsOf(concepts), Collections.emptySet()).getSync();
+				.prepareGetApplicablePredicates(MAIN_BRANCH, idsOf(concepts), ancestorsOf(concepts), Collections.emptySet(), relationshipsOf(concepts)).getSync();
+	}
+
+	private Set<String> relationshipsOf(SnomedConcepts concepts) {
+		return concepts.getItems()
+				.stream()
+				.flatMap(concept -> concept.getRelationships().stream().map(relationship -> relationship.getTypeId() + Ecl.EQUAL + relationship.getDestinationId()))
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.api.rest.tests/src/com/b2international/snowowl/snomed/api/japi/io/SnomedRefSetDSVExportTest.java]: The format of a relationship MRCM domain should be moved to the `SnomedConstraintDocument` class.
For example: `SnomedConstraintDocument.relationshipDomainOf(typeId, destinationId)`.

**Review by cmark — CHANGES_REQUESTED** [review]

File: snomed/com.b2international.snowowl.snomed.api.rest.tests/src/com/b2international/snowowl/snomed/api/japi/io/SnomedRefSetDSVExportTest.java [hunk 3]
```diff
@@ -180,7 +181,14 @@ public void mapTypeDSVExport() throws Exception {
...
 	private Iterable<SnomedConstraint> getConstraints(SnomedConcepts concepts) {
 		return SnomedRequests
-				.prepareGetApplicablePredicates(MAIN_BRANCH, idsOf(concepts), ancestorsOf(concepts), Collections.emptySet()).getSync();
+				.prepareGetApplicablePredicates(MAIN_BRANCH, idsOf(concepts), ancestorsOf(concepts), Collections.emptySet(), relationshipsOf(concepts)).getSync();
+	}
+
+	private Set<String> relationshipsOf(SnomedConcepts concepts) {
+		return concepts.getItems()
+				.stream()
+				.flatMap(concept -> concept.getRelationships().stream().map(relationship -> SnomedConstraintDocument.relationshipDomainOf(relationship.getTypeId(), relationship.getTypeId())))
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.api.rest.tests/src/com/b2international/snowowl/snomed/api/japi/io/SnomedRefSetDSVExportTest.java]: In `SnomedConstraintDocument.relationshipDomainOf(...)`, the second parameter should be the relationship's destination ID.

**Review by apeteri — CHANGES_REQUESTED** [review]

**Comment by apeteri** [comment]
Changes included in #236, closing PR.

**Pull request closed without merging**